integrations/June/browser.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable */
+import logger from "../../utils/logUtil";
+import get from "get-value";
+import { NAME } from "./constants";
+import { LOAD_ORIGIN } from "../ScriptLoader";
+
+class June {
+  constructor(config) {
+    this.name = NAME;
+    this.apiKey = config.apiKey;
+  }
+
+  loadScript() {
+    window.analytics = {};
+    function juneify(writeKey) {
+      window.analytics._writeKey = writeKey;
+      const script = document.createElement("script");
+      script.type = "application/javascript";
+      script.src =
+        "https://unpkg.com/@june-so/analytics-next/dist/umd/standalone.js";
+      script.setAttribute("data-loader", LOAD_ORIGIN);
+      const first = document.getElementsByTagName("script")[0];
+      first.parentNode.insertBefore(script, first);
+    }
+    juneify(this.apiKey);
+  }
+
+  init() {
+    logger.debug("===In init June===");
+    this.loadScript();
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded June===");
+    return !!window.analytics && typeof window.analytics === "object";
+  }
+
+  isReady() {
+    logger.debug("===In isReady June===");
+    return !!window.analytics;
+  }
+
+  page(rudderElement) {
+    logger.debug("===In June page===");
+    const { name, properties } = rudderElement.message;
+    window.analytics.page(name, properties);
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In June identify===");
+    const { message } = rudderElement;
+    const userId =
+      get(message, "userId") ||
+      get(message, "context.traits.userId") ||
+      get(message, "context.traits.Id");
+    const traits = get(message, "context.traits");
+    if (!userId) {
+      logger.error("userId is required for an identify call");
+      return;
+    }
+    window.analytics.identify(userId, traits);
+  }
+
+  track(rudderElement) {
+    logger.debug("===In June track===");
+    let groupId;
+    const { event } = rudderElement.message;
+    let { properties } = rudderElement.message;
+    ({ groupId, ...properties } = properties || {});
+
+    if (groupId) {
+      window.analytics.track(event, properties, { groupId });
+    } else {
+      window.analytics.track(event, properties);
+    }
+  }
+
+  group(rudderElement) {
```

Comment:
aren't we supposed to add user to the group if the data is available? In that case we need to fetch data from `context.traits`. The data we are fetching now , ie root level traits are the traits of the particular group only.

Reply:
also somehow the userId can come into the picture, because someone may want to add old/new member to the group

Reply:
It's imperative to call user identification call before group call to create a company successfully .For company details updation, identification call is optional.
We can't pass userId in group call. 
The june `group` call associate the current user from `identify` call with a company.